keps/sig-auth/4317-pod-certificates/README.md

```diff
@@ -0,0 +1,1046 @@
+identity information into an X.509 certificate or certificate request, we can
+define a new ASN.1 OID (currently `1.3.6.1.4.1.57683.1`, though we need to
+define a management policy for the PEN) and specify that it contains the
+following DER-encoded structure:
+
+```go
+type podIdentityASN1 struct {
+	Namespace          string `asn1:"utf8"`
+	ServiceAccountName string `asn1:"utf8"`
+	PodName            string `asn1:"utf8"`
+	PodUID             string `asn1:"utf8"`
+	NodeName           string `asn1:"utf8"`
+}
+```
+
+Certificates and CertificateRequests that embed this extension do not need to include any data in the standard X.509 Name or Subject Alternate Name fields.  The CertificateRequests generated by Kubelet will not, nor will the certificates issued by `kubernetes.io/kube-apiserver-client-pod`.
+
+Utility functions for embedding and extracting this extension from X.509
+Certificate and CertificateRequest objects are defined in
+`k8s.io/component-helpers/kubernetesx509`.  They are intended to be used both by
+core Kubernetes components and third-party software that needs to interact with
+these certificates.
+
+### PodIdentity Extension Node Restriction
+
+The noderestriction admission plugin will check the consistency of PodIdentity extensions in CertificateSigningRequest objects created by `system:node:` identities.  The creation request will only be admitted if the PodIdentity extension refers to a pod that is currently running on the node that requested the certificate.
+
+This upholds the Kubernetes node isolation guarantee by ensuring that
+compromising a single node only results in compromises of workloads that have
+actually been scheduled onto that node.
+
+<<[UNRESOLVED]>>
+Is it necessary or desirable to forbid non-Kubelet identities from creating CertificateSigningRequest objects that embed a PodIdentity extension?
+<<[/UNRESOLVED]>>
+
+During Alpha, this capability is behind the PodIdentityX509Support feature gate.
+
+### PodCertificate Projected Volume Sources
+
+The PodCertificate projected volume source instructs Kubelet to inject a
+credential bundle consisting of a private key and certificate chain into the
+container fileystem.  Kubelet handles generating the key, requesting a
+certificate from the named signer, and renewing the certificate before it
+expires.
+
+Kubelet supports only a few key generation strategies, each named by a string
+that encapsulates both the key type and any parameters necessary (for example,
+RSA modulus size).  The intention is to offer a a tasting menu of reasonable key
+choices, rather than offering a flexibility to a wide variety of parameters.
+Key types supported by Kubelet are "RSA2048", "RSA3072", "RSA4096", "ECDSAP256",
+and "ECDSAP384".  If no key type is specified, Kubelet defaults to "ECDSAP256".
+
+<<[UNRESOLVED]>>
+Should RSA2048 be offered?  I'm inclined to say no.  Are there any other key types that should be offered?
```

Comment:
“I” isn't always clear - it's helpful to name who objects.

Reply:
Done.